job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/step/StepContext.groovy

```diff
@@ -42,6 +42,26 @@ class StepContext extends AbstractExtensibleContext {
     }
 
     /**
+     * Runs a xshell script.
+     *
+     * @since 1.41
+     */
+    @RequiresPlugin(id = 'xshell', minimumVersion = '0.10')
+    void xshell(String commandLine = null, Boolean executeFromWorkingDir = false, String regexToKill = null,
```

Comment:
Replace the parameters by a nested context. Contexts are more convenient for users and allow to specify selected parameters.